tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -609,6 +609,15 @@ def do_testharness(self, protocol, url, timeout):
         while True:
             result = protocol.base.execute_script(
                 self.script_resume, asynchronous=True, args=[strip_server(url)])
+            # A user prompt created after starting execution of the resume
+            # script will cause WebDriver to return `null` here [1, 2]. In that
+            # case, cycle this event loop and handle the prompt the next time
+            # the resume script executes.
+            #
+            # [1]: Step 5.3 of https://www.w3.org/TR/webdriver/#execute-async-script
+            # [2]: https://www.w3.org/TR/webdriver/#dfn-execute-a-function-body
+            if result is None:
```

Comment:
How should this interact with L626? Do we have other cases result will be None, and re-execute `script_resume` will be correct for those cases also?

Reply:
> How should this interact with L626?

Thanks for pointing this out. Fixed so that L626 (browser liveness check) happens first; otherwise, it would have been a no-op.

> Do we have other cases result will be None

During normal operation, the resume script's `resolve()` callback is only ever called [with the three-arg payload](https://github.com/web-platform-tests/wpt/blob/4ee1931a4a7993cf089ebfe83a2548f0aea2b876/tools/wptrunner/wptrunner/testharnessreport.js#L67-L69). I found one [other exception in the spec (step 4)](https://www.w3.org/TR/webdriver/#dfn-execute-a-function-body) when the resume script is [not valid JS](https://262.ecma-international.org/5.1/#sec-16), but I don't think we need to consider that.

Reply:
There could also be cases that are implementation specific. I think we should comment in a way that handling alert's is one reason we should do a retry.

Reply:
> There could also be cases that are implementation specific.

Yeah. I think I prematurely jumped to the conclusion that `alert()` was causing all such `ERROR`s on Chrome on iOS. Jie pointed out there are [some tests](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/html/webappapis/timers/clearinterval-from-callback.any.js?q=f:external%2Fwpt%2Fhtml%2Fwebappapis%2Ftimers%2Fclearinterval-from-callback.any.js) failing the [same way](https://chromium-swarm.appspot.com/task?id=67c5c5b74ca35010) that don't use prompts, so retrying the testdriver loop may not guarantee (timely) termination.

I'd be happy to submit just the test in that case and drop the executor "fix" from this PR.

> I think we should comment in a way that handling alert's is one reason we should do a retry.

That's what the comment says, just with the more general term "user prompt" and not "alert"?